src/python/ensembl/production/factories/metadata/genomeFactory.py

```diff
@@ -140,13 +161,13 @@ def resolve_genome_list(self, info, filters=None):
                 query = query.filter(OrganismGroup.name.in_(filters.organism_group))
 
             if filters.organism_name:
-                ensembl_names = set(filters.organism_name) - set(filters.anti_organism_name)
-                if ensembl_names:
-                    query = query.filter(Genome.organism.has(Organism.ensembl_name.in_(ensembl_names)))
+                biosample_id = set(filters.organism_name) - set(filters.anti_organism_name)
```

Comment:
All this is to be able to filter by which values. I ask because initially the ensembl_name was the production_name, then it became cumbersome to generate those for GBuilders. So we moved away to find something which is representing something real. But I wonder if this is the implemented/intended beheviour here. e.g. if this is to force to process one particular DB, then the test would be done against genome.production_name instead.

Reply:
Yes, you are right we want to filter the genome uuids concerning the production name &  will also include for filter for biosample id. thanks for spotting this